scripts/gen_mmu.py

```diff
@@ -0,0 +1,538 @@
+import os
+import sys
+import struct
+import parser
+from collections import namedtuple
+import ctypes
+import argparse
+
+############# global variables
+pd_complete = ''
+inputfile = ''
+outputfile = ''
+list_of_pde = {}
+num_of_regions = 0
+read_buff=''
+
+struct_mmu_regions_tuple = {"start_addr","size","permissions"}
+mmu_region_details = namedtuple("mmu_region_details", "pde_index page_entries_info")
+
+valid_pages_inside_pde = namedtuple("valid_pages_inside_pde","start_addr size \
+                                pte_valid_addr_start \
+                                pte_valid_addr_end \
+                                permissions")
+
+page_tables_list = []
+pd_start_addr = 0
+validation_issue_memory_overlap = [False, 0, -1]
+output_offset = 0
+print_string_pde_list = ''
+pde_pte_string = {}
+
+#############
+
+#return the page directory number for the give address
+def get_pde_number(value):
+    return( (value >> 22 ) & 0x3FF)
+
+#return the page table number for the given address
+def get_pte_number(value):
+    return( (value >> 12 ) & 0x3FF)
+
+
+#consider little endian
+def read_little_endian_binay(file_name):
```

Comment:
"binary" not "binay"
Actually this function seems unused? You should be using struct module for this